### PR TITLE
Control plane security group always whitelist worker security group and revert #186

### PR DESCRIPTION
**Pull request by ryanooi**
# PR o'clock

## Description

1. Main change on cluster.tf `aws_security_group_rule.cluster_https_worker_ingress` intended to whitelist port 443 for newly created security group or self attach security group from worker's security group. Self attach control plane security group will not able to whitelist worker security group before it get created therefore additional step will required.
2. Revert PR #186 as Terraform v0.12.x fixed "count cannot be computed" issue
2.1  Remove `cluster_create_security_group` and `worker_create_security_group` to improve consistency. There is no guarantee/enforcement when you set `cluster_create_security_group` or `worker_create_security_group` to `false`, user will also provide `cluster_security_group_id` or `worker_security_group_id`. 

### Checklist

- [ ] Change added to CHANGELOG.md. All changes must be added and breaking changes and highlighted
- [x] CI tests are passing
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by ryanooi** [comment]
Need some explanation? or?

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ project adheres to [Semantic Versioning](http://semver.org/).
 - Updated minimum version of Terraform to avoid a bug (by @dpiddockcmp)
 - Fix cluster_oidc_issuer_url output from list to string (by @chewvader)
 - Fix idempotency issues for node groups with no remote_access configuration (by @jeffmhastings)
+- Improve UX of create new EKS cluster (by @ryanooi)
```
Comment (max-rocket-internet) [inline CHANGELOG.md]: Can you update this to describe your changes more accurately? i.e. "changed logic around SG and SG rule creation"

**Comment by max-rocket-internet** [comment]
Actually, this is a breaking change as you are removing some vars. So also please write details under `Important notes` that clearly explain how to migrate after this change 🙂

File: CHANGELOG.md (hunk 1 above)
Comment (ryanooi) [reply CHANGELOG.md]: Updated. Please help review again :)

**Pull request merged**